Mathlib/RingTheory/Nilpotent.lean

```diff
@@ -202,7 +228,20 @@ end Commute
 
 section CommSemiring
 
-variable [CommSemiring R]
+variable [CommSemiring R] {x y : R}
+
+lemma isNilpotent_sum {ι : Type _} {s : Finset ι} {f : ι → R}
+    (hnp : ∀ i ∈ s, IsNilpotent (f i)) :
+    IsNilpotent (s.sum f) :=
+  Commute.isNilpotent_sum hnp fun _ _ _ _ ↦ Commute.all _ _
+
+@[simp] lemma isNilpotent_mul_left_iff [NoZeroDivisors R] (hy : y ≠ 0) :
+    IsNilpotent (x * y) ↔ IsNilpotent x :=
```

Comment:
I'm reviewing on mobile, so I may be misunderstanding.  NoZeroDivisors implies in particular that only `0` is nilpotent, right?

Reply:
Absolutely, this was a thinko on my part, caused by mechanical generalisation.